src/openpgp/packet/keymaterial.iced

```diff
@@ -479,7 +515,7 @@ class KeyMaterial extends Packet
         expire_at = generated + expire_in
         if not winner? or (winner.expire_at? and (winner.expire_at < expire_at))
           winner = { expire_at, generated, expire_in }
-      else if (expire_in? and expire_in is 0) or (not expire_in? and not @is_primary())
```

Comment:
Remind me why it wouldn't have worked to just delete `is_primary` here. What case did that break?